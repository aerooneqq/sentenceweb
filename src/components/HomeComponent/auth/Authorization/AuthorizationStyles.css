#signInInputContainer {
    width: 80%;
    margin-left: 10%;
    background: #f0f0f0;
    display: flex;
    flex-direction: column;

    animation: authContentRotation 1s ease forwards
}

    @keyframes authContentRotation {
        from {
            visibility: hidden;
            transform: rotate3d(0, 1, 0, 180deg)
        }

        100% { 
            transform: rotate3d(0, 1, 0, 360deg)
        }
    }
  
.inputPropContainer {
    margin-top: 25px;
}
    
.inputNameText {
    font-family: Arial;
    font-size: 15px;
    color: #606060;
    letter-spacing: 0px;
    font-weight: bold;
}

.authInput {
    width:90%;
    height: 50px;
  
    margin-top: -10px;
  
    border-style: solid;
    border-width: 1px;
    border-color: orange;
    border-radius: 5px;
  
    font-size: 21px;
    font-family: Arial;
  
    padding-inline-start: 5%;
    padding-inline-end: 5%;
}
  
.authInput:hover {
    border-color: orangered;
    outline: none;
}
  
.authInput:focus {
    border-color: orangered;
    outline: none;
}
  
#btnContainer {
    margin-top: 40px;
}
  
    #btnContainer p {
        font-family: Arial;
        font-size: 15px;
        color: gray;
        cursor: default;
    }
    
    #btnContainer p:hover {
        text-decoration: underline;
    }

#signInBtn {
    width: 100%;
    height: 60px;
    background: linear-gradient(32deg, orange, orangered);
  
    margin-top: -5px;
  
    border-style: none;
    border-radius: 5px;
  
    font-family: Arial;
    color: white;
    font-size: 22px;
}
  
#signInBtn:hover {
    background: linear-gradient(32deg, #FD8A60, #FC5B20);
    outline: none;
}

#signInBtn:focus {
    outline: none;
}

#authorizationLoaderCont { 
    margin-top: 100px;
  
    display: flex;
    align-items: center;
    justify-content: center;
}


  

  