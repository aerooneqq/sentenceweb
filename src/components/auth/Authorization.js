import React from "react"
import "./RegAuthStyles.css"

export default class Authorization extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
          email: "",
          password: ""
        };

        this.handleEmailInputChange = this.handleEmailInputChange.bind(this)
        this.handlePasswordInputChange = this.handlePasswordInputChange.bind(this)
        this.handleSignIn = this.handleSignIn.bind(this);
    }

    componentDidMount(){
      var authCont = document.getElementById("signInInputContainer")
      authCont.classList.toggle("authContentRotationThreeD")
    }

    handleEmailInputChange(event){
      this.setState({
        email: event.target.value
      })
    }

    handlePasswordInputChange(event){
      this.setState({
        password: event.target.value
      })
    }

    handleSignIn(){
      let email = this.state.email;
      let password = this.state.password;

      this.tryToAuthorizeUser({email: email, password: password})
    }

    tryToAuthorizeUser(userInput){
      //Performing an hhtp call here in future
      let user = {email: "aerooneQ@yandex.ru", name: "Aero"}
      localStorage.setItem("token", "a")

      alert(localStorage.getItem("token"))
    }

    render() {
        return (
          <div id='signInInputContainer'>
            <div className="inputPropContainer">
              <p className="inputNameText">E-mail</p>
              <input type='text' value={this.state.email} id="inputEmail" className="authInput"
                    onChange={this.handleEmailInputChange}/>
            </div>
            <div className="inputPropContainer">
              <p className="inputNameText">Password</p>
              <input type='password' value={this.state.password} id='inputPassword' className="authInput"
                    onChange={this.handlePasswordInputChange}/>
            </div>
            <div id="btnContainer">
              <p>Forgot your password?</p>
              <button id="signInBtn" onClick={this.handleSignIn}>Sign in</button>
            </div>
          </div>
       );
    }
}
