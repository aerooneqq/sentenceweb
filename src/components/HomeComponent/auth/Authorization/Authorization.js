import React, {Component, lazy} from "react"

//Styles
import "./AuthorizationStyles.css"

//Components
const Loader = lazy(()=>import("../../../loader/Loader"))

export default class Authorization extends Component {
    constructor(props) {
        super(props);

        this.state = {
          email: "",
          password: "",
          isAuthorizing: false
        };

        this.handleEmailInputChange = this.handleEmailInputChange.bind(this)
        this.handlePasswordInputChange = this.handlePasswordInputChange.bind(this)
        this.handleSignIn = this.handleSignIn.bind(this);
    }

    componentDidMount(){
      var authCont = document.getElementById("signInInputContainer")
      authCont.classList.toggle("authContentRotationThreeD")
    }

    handleEmailInputChange(event){
      this.setState({
        email: event.target.value
      })
    }

    handlePasswordInputChange(event){
      this.setState({
        password: event.target.value
      })
    }
    
    handleSignIn(){
      let email = this.state.email;
      let password = this.state.password;

      this.setState({
        isAuthorizing: true
      });

      this.props.signIn(email, password).then(() => {
        this.setState({
          isAuthorizing: false
        })
      }).catch(() => { 
        this.setState({
          isAuthorizing: false
        })
      })
    }

    render() {
        if (this.state.isAuthorizing) { 
          return(
            <div id = "authorizationLoaderCont">
              <Loader />
            </div>
          )
        }
        else{ 
          return (
            <div id='signInInputContainer'>
              <div className="inputPropContainer">
                <p className="inputNameText">E-mail</p>
                <input type='text' value={this.state.email} id="inputEmail" className="authInput"
                      onChange={this.handleEmailInputChange}/>
              </div>
              <div className="inputPropContainer">
                <p className="inputNameText">Password</p>
                <input type='password' value={this.state.password} id='inputPassword' className="authInput"
                      onChange={this.handlePasswordInputChange}/>
              </div>
              <div id="btnContainer">
                <p>Forgot your password?</p>
                <button id="signInBtn" onClick={this.handleSignIn}>Sign in</button>
              </div>
            </div>
         );
        }
    }
}
