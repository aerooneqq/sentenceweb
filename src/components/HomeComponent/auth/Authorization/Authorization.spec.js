import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import Authorization from './Authorization';

describe("Authorization", () => { 
    it("must render correctly", () => { 
        const props = { 
            signIn: () => {}
        }

        const authorization = shallow(<Authorization {...props}/>);

        expect(shallowToJson(authorization)).toMatchSnapshot()
    });

    it("must have right text", () => { 
        const props = { 
            signIn: () => {}
        }

        const authorization = shallow(<Authorization {...props}/>);

        expect(authorization.find("p.inputNameText").at(0).text()).toBe("E-mail");
        expect(authorization.find("p.inputNameText").at(1).text()).toBe("Password");
        expect(authorization.find("p").at(2).text()).toBe("Forgot your password?");
        expect(authorization.find("button").text()).toBe("Sign in");
    });

    it("must call sign in method when sign in btn clicked", () => {
        let isSignInCalled = false; 
        const props = { 
            signIn: (email, pass) => { 
                return new Promise((onResolved, onRejected) => {
                    isSignInCalled = true;
                });
            }
        }

        const authorization = shallow(<Authorization {...props} />);

        authorization.find("button").simulate("click");
        expect(isSignInCalled).toBe(true);
    });

    it("must show loader when is loading", () => { 
        const authorization = shallow(<Authorization />);

        authorization.setState({ 
            isAuthorizing: true
        });

        expect(authorization.find("div").at(0).hasClass("authorizationLoaderCont")).toBe(true)
    })
});