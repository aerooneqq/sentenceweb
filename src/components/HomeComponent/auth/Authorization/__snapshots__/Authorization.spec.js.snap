// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization must render correctly 1`] = `
<div
  className="authContentRotationThreeD"
  id="signInInputContainer"
>
  <div
    className="inputPropContainer"
  >
    <p
      className="inputNameText"
    >
      E-mail
    </p>
    <input
      className="authInput"
      id="inputEmail"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="inputPropContainer"
  >
    <p
      className="inputNameText"
    >
      Password
    </p>
    <input
      className="authInput"
      id="inputPassword"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value=""
    />
  </div>
  <div
    id="btnContainer"
  >
    <p>
      Forgot your password?
    </p>
    <button
      id="signInBtn"
      onClick={[Function]}
    >
      Sign in
    </button>
  </div>
</div>
`;
